extensions/blocks/social-previews/upgrade.js

```diff
@@ -1,15 +1,36 @@
 /**
  * External dependencies
  */
+import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import { Button } from '@wordpress/components';
+import analytics from '../../../_inc/client/lib/analytics';
 import upgradeImageUrl from './upgrade-illustration.svg';
+import { BUSINESS_PLAN } from '../../shared/components/upgrade-nudge/constants';
+import useUpgradeFlow from "../../shared/use-upgrade-flow";
+
+const trackViewEvent = () =>
+	void analytics.tracks.recordEvent( 'jetpack_editor_block_upgrade_nudge_impression', {
+		plan: BUSINESS_PLAN,
+		block: 'social-previews',
+	} );
+
+const trackClickEvent = () =>
+	void analytics.tracks.recordEvent( 'jetpack_editor_block_upgrade_click', {
+		plan: BUSINESS_PLAN,
+		block: 'social-previews',
+	} );
+
+export default function SocialPreviewsUpgrade() {
+	const [ href, autosaveAndRedirect ] = useUpgradeFlow( BUSINESS_PLAN, trackClickEvent );
```

Comment:
we shouldn't hardcode the required plan. It should be picked up dynamically from the Jetpack global initial state object.

Reply:
Fixed in 864050b